### PR TITLE
[DNM] EQ IIR conversion to use module interface

### PR DESCRIPTION
**Pull request by ranj063**

### CODE REVIEW

**Comment by ranj063** [comment]
SOFCI TEST

File: src/audio/eq_iir/eq_iir.c [hunk 1]
```diff
@@ -444,11 +480,14 @@ static void eq_iir_free_delaylines(struct comp_data *cd)
...
-static int eq_iir_init_coef(struct sof_eq_iir_config *config,
-			    struct iir_state_df2t *iir, int nch)
+static int eq_iir_init_coef(struct processing_module *mod, int nch)
 {
+	struct comp_data *cd = module_get_private_data(mod);
+	struct sof_eq_iir_config *config = cd->config;
+	struct iir_state_df2t *iir = cd->iir;
 	struct sof_eq_iir_header_df2t *lookup[SOF_EQ_IIR_MAX_RESPONSES];
 	struct sof_eq_iir_header_df2t *eq;
+	struct comp_dev *dev = mod->dev;
```
Comment (kv2019i) [inline src/audio/eq_iir/eq_iir.c:489]: @ranj063 So please remove this if dev is only used for debugs, and instead:

File: src/audio/eq_iir/eq_iir.c [hunk 2]
```diff
@@ -457,19 +496,18 @@ static int eq_iir_init_coef(struct sof_eq_iir_config *config,
 	int j;
 	int s;
 
-	comp_cl_info(&comp_eq_iir, "eq_iir_init_coef(), response assign for %u channels, %u responses",
-		     config->channels_in_config,
-		     config->number_of_responses);
+	comp_info(dev, "eq_iir_init_coef(), response assign for %u channels, %u responses",
```
Comment (kv2019i) [inline src/audio/eq_iir/eq_iir.c:498]: @ranj063 Have "mod->dev" here. We had this prior to the module conversion, gcc was upgraded during this time and now your changes are bringing the warnings back.

Comment (ranj063) [reply src/audio/eq_iir/eq_iir.c]: Thanks @kv2019i

File: src/audio/CMakeLists.txt [hunk 3]
```diff
@@ -208,7 +208,7 @@ set(mixer_sources ${mixer_src})
 set(src_sources src/src.c src/src_generic.c)
 set(asrc_sources asrc/asrc.c asrc/asrc_farrow.c asrc/asrc_farrow_generic.c)
 set(eq-fir_sources eq_fir/eq_fir.c eq_fir/eq_fir_generic.c)
-set(eq-iir_sources eq_iir/eq_iir.c)
+set(eq-iir_sources module_adapter/module_adapter.c module_adapter/module/generic.c eq_iir/eq_iir.c)
```
Comment (lyakh) [inline src/audio/CMakeLists.txt:211]: don't we now add these in src/audio/module_adapter/CMakeLists.txt?

File: src/audio/data_blob.c [hunk 4]
```diff
@@ -476,10 +476,8 @@ comp_data_blob_handler_new_ext(struct comp_dev *dev, bool single_blob,
...
-	if (!blob_handler)
-		return;
-
-	comp_free_data_blob(blob_handler);
-
-	rfree(blob_handler);
+	if (blob_handler) {
+		comp_free_data_blob(blob_handler);
+		rfree(blob_handler);
+	}
```
Comment (lyakh) [inline src/audio/data_blob.c:482]: sorry, I don't think this is a good patch or at least a good commit description. I understand that compilers have their problems and sometimes small and seemingly unrelated changes might change the compiler behaviour, but I don't think we should rely on that. I don't think we can rely on apparent empiric improvements which have no logical explanation. If our success rate depends on the if-else ordering then (arguably) we are doing something wrong. I'd propose to drop this commit and investigate the problem.

**Review by lyakh — CHANGES_REQUESTED** [review]

File: src/audio/data_blob.c (hunk 4 above)
Comment (ranj063) [reply src/audio/data_blob.c]: yes, sorry this patch needs to go. It was there for a test for xtos and not needed with zephyr.

**Comment by gkbldcig** [comment]
Can one of the admins verify this patch?

**Comment by singalsu** [comment]
@ranj063 A new version of this PR is in #6992. I will add same improvements to run-time switch bypass/process and IPC4 as I did for FIR.

File: src/audio/CMakeLists.txt (hunk 3 above)
Comment (kv2019i) [reply src/audio/CMakeLists.txt:211]: ... there's indeed some CONFIG_COMP_FIR logic in module_adapter CMakeLists.txt.

File: src/include/sof/audio/module_adapter/module/generic.h [hunk 5]
```diff
@@ -209,4 +209,29 @@ int module_get_large_config(struct comp_dev *dev, uint32_t param_id, bool first_
 			    bool last_block, uint32_t *data_offset, char *data);
 int module_adapter_get_attribute(struct comp_dev *dev, uint32_t type, void *value);
 
+static inline void
+module_update_processing_position(struct input_stream_buffer *input,
+				  struct output_stream_buffer *output,
+				  enum sof_ipc_frame input_frame_fmt,
+				  enum sof_ipc_frame output_frame_fmt, int samples)
+{
```
[the remark this answers is not in the document]
Comment (kv2019i) [reply src/include/sof/audio/module_adapter/module/generic.h:217]: This is potentially ok, but needs to be confirmed that compiler can generate good code for this construct.

**Comment by singalsu** [comment]
> @singalsu does this PR need to be closed ?

Yes, I don't think here's anything that would add value on top of #6992. I tried to copy everything relevant from here.

**Pull request closed without merging**